api/v1alpha1/promotion_types.go

```diff
@@ -64,13 +74,37 @@ const (
 	PromotionStepStatusSkipped PromotionStepStatus = "Skipped"
 )
 
-// IsTerminal returns true if the PromotionPhase is a terminal one.
-func (p *PromotionPhase) IsTerminal() bool {
-	switch *p {
-	case PromotionPhaseSucceeded, PromotionPhaseFailed, PromotionPhaseErrored, PromotionPhaseAborted:
-		return true
+// Compare compares the severity of the current PromotionStepStatus with
+// another. It returns -1 if the current status is less severe than the other, 0
+// if the two statuses are equally severe, or 1 if the current status is more
+// severe than the other.
+func (s PromotionStepStatus) Compare(rhs PromotionStepStatus) int {
+	// Define severity levels for each status
+	severity := map[PromotionStepStatus]int{
+		PromotionStepStatusSucceeded: 0,
+		PromotionStepStatusSkipped:   1,
+		PromotionStepStatusRunning:   2,
+		PromotionStepStatusAborted:   3,
+		PromotionStepStatusFailed:    4,
+		PromotionStepStatusErrored:   5,
+	}
```

Comment:
I think this can be moved outside the method.

Reply:
I'd meant to and forgot. I didn't intend to re-instantiate this map every time. Fixing.